pkg/devfile/parse.go

```diff
@@ -0,0 +1,61 @@
+package devfile
+
+import (
+	"github.com/devfile/parser/pkg/devfile/parser"
+)
+
+// This is the top level parse code which has validation code specific to odo hence it cannot be kept inside the
```

Comment:
why we need this here? I think it could be removed, later on we need to think how we want to expose library functions.